src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c

```diff
@@ -21,77 +42,697 @@ const ActorInit En_Goroiwa_InitVars = {
+        sp8C = this->unk_1BC;
+    }
+    sp8C *= this->unk_1C0;
+    // temp assignment needed to match
+    temp = &sp80;
+    if (this->unk_1D3 & 8) {
+        func_80A4BF28(this, globalCtx, &sp28);
+        Math3D_Vec3f_Cross(&unitY, &this->actor.velocity, temp);
+    } else {
+        Math3D_Vec3f_Cross(&unitY, &this->actor.velocity, temp);
+    }
+
+    if (EnGoroiwa_Vec3fNormalize(&sp74, temp)) {
+        this->unk_1B0 = sp74;
+    } else {
+        sp74 = this->unk_1B0;
+    }
+
+    func_800D23FC(sp8C, &sp74, 0);
+    Matrix_RotateY(this->actor.shape.rot.y * (2.0f * M_PI / 0x10000), 1);
+    Matrix_RotateX(this->actor.shape.rot.x * (2.0f * M_PI / 0x10000), 1);
+    Matrix_RotateZ(this->actor.shape.rot.z * (2.0f * M_PI / 0x10000), 1);
+    Matrix_Get(&mtx);
+    func_800D20CC(&mtx, &this->actor.shape.rot, 0);
+}
+
+void func_80A4D074(EnGoroiwa* this, GlobalContext* globalCtx) {
+    s16 temp_v0 = (this->actor.params >> 8) & 3;
+
+    func_80A4C134(this);
+
+    if (temp_v0 == 0 || temp_v0 == 1) {
+        if (this->currentWaypoint == 0 || this->currentWaypoint == this->endWaypoint) {
+            func_80A4C1C4(this, globalCtx, this->currentWaypoint);
+        }
+    }
+
+    func_80A4BE54(this, globalCtx);
+}
+
+void func_80A4D0FC(EnGoroiwa* this, GlobalContext* globalCtx) {
+    static f32 colliderHeightOffset[] = { 0.0f, 59.5f };
+
+    s16 angle1;
+    s16 angle2;
+    s32 pad;
+    Vec3f* thisPos = &this->actor.posRot.pos;
+    Vec3f effectPos;
+    Vec3f effectVelocity;
+    f32 temp_f24;
+    f32 temp_f22;
+    f32 temp_f20;
+    f32 temp_f20_2;
+    s16 temp_v0 = (this->actor.params >> 10) & 1;
+    s32 i;
+
+    for (i = 0, angle1 = 0; i < 16; i++, angle1 += 0x4E20) {
+        temp_f22 = Math_Sins(angle1);
+        temp_f24 = Math_Coss(angle1);
+        angle2 = Math_Rand_ZeroOne() * 0xFFFF;
+        effectPos.x = ((Math_Rand_ZeroOne() * 50.0f) * temp_f22) * Math_Sins(angle2);
+        temp_f20_2 = Math_Sins(angle2);
+        effectPos.y = (((Math_Rand_ZeroOne() - 0.5f) * 100.0f) * temp_f20_2) + colliderHeightOffset[temp_v0];
+        effectPos.z = ((Math_Rand_ZeroOne() * 50.0f) * temp_f24) * Math_Sins(angle2);
+        effectVelocity.x = effectPos.x * 0.2f;
+        effectVelocity.y = (Math_Rand_ZeroOne() * 15.0f) + 2.0f;
+        effectVelocity.z = effectPos.z * 0.2f;
+        Math_Vec3f_Sum(&effectPos, thisPos, &effectPos);
+        func_80029E8C(globalCtx, &effectPos, &effectVelocity, &effectPos, -340, 33, 28, 2, 0,
+                      (Math_Rand_ZeroOne() * 7.0f) + 1.0f, 1, 0, 70, -1, 1, D_0400D340);
+    }
+
+    effectPos.x = thisPos->x;
+    effectPos.y = thisPos->y + colliderHeightOffset[temp_v0];
+    effectPos.z = thisPos->z;
+    func_80033480(globalCtx, &effectPos, 80.0f, 5, 70, 110, 1);
+    func_80033480(globalCtx, &effectPos, 90.0f, 5, 110, 160, 1);
+}
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32_DIV1000(gravity, 64676, ICHAIN_CONTINUE), ICHAIN_F32_DIV1000(minVelocityY, 50536, ICHAIN_CONTINUE),
```

Comment:
`gravity` is more likely `-860` or something close if I got the math off,
and `minVelocityY` `-15000`